### ISSUE TITLE
jax.scipy.optimize.minimize ignores TOL setting

### ISSUE THREAD

**Issue by agudym** [google/jax#19122]
### Description

Hi, guys!

Changing the TOL parameter doesn't affect the result (while for scipy-version it does a lot):

```python
import numpy as np

from scipy.optimize import minimize as scipy_minimize

from jax.scipy.optimize import minimize as jax_minimize
from jax import config
config.update("jax_enable_x64", True)

params_gt = np.array((0.1,))
params_init = np.array((1.,))

def f(x):
    return (x - params_gt)[0]**4

for tol in (None, 1e-6, 1e-12):
    print(f"TOL {tol}")
    for method, desc in zip((scipy_minimize, jax_minimize), ("SCIPY", "JAX")):
        res = method(f, params_init, method="BFGS", tol=tol )
        print(f"\t{desc}: x={res.x} f(x)={res.fun} iteration={res.nit}")

#TOL None
#	SCIPY: x=[0.08903354] f(x)=1.4463248855556928e-08 iteration=10
#	JAX: x=[0.08834503] f(x)=1.8452031122761725e-08 iteration=10
#TOL 1e-06
#	SCIPY: x=[0.09375078] f(x)=1.5251138322835735e-09 iteration=12
#	JAX: x=[0.08834503] f(x)=1.8452031122761725e-08 iteration=10
#TOL 1e-12
#	SCIPY: x=[0.09994754] f(x)=7.571085068707287e-18 iteration=29
#	JAX: x=[0.08834503] f(x)=1.8452031122761725e-08 iteration=10
```

The parameter affects termination criteria's gradient length ("gtol"), and significantly affects solutions quality. 

### What jax/jaxlib version are you using?

0.4.23

### Which accelerator(s) are you using?

CPU/GPU, doesn't matter

### Additional system info?

Ubuntu 22.04, x86, Python 3.10.12

**Comment by jakevdp** [comment]
Thanks for the report! It looks like you're right that the `tol` argument to `jax.scipy.optimize.minimize` is not referenced. Unfortunately, this is not something we will probably fix, because we've decided to deprecate `jax.scipy.optimize` in favor of better optimizer implementations available in the [jaxopt](Does scan with no carry effectively reduce to vmap?) package. I'd suggest using that package if you're doing optimization in JAX.

**Comment by agudym** [comment]
I see, thank you!

Actually so far I'm sticking to a "mixed" scipy-jax approach, i.e. in many problems Jacobian-computation is the main issue, and JAX is insanely efficient in that, while the "main" loop is implemented by high-level scipy-code, here is the benchmark results:

```
NUMPY:
64.7 ms ± 3.36 ms per loop (mean ± std. dev. of 7 runs, 10 loops each)
JAX:
3.77 ms ± 258 µs per loop (mean ± std. dev. of 7 runs, 1 loop each)
MIX:
11.2 ms ± 1.23 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)
```

```python
import matplotlib.pyplot as plt

import numpy as np
from scipy.optimize import minimize as scipy_minimize
from scipy.special import erf as scipy_erf

import jax
from jax.scipy.optimize import minimize as jax_minimize
from jax.scipy.special import erf as jax_erf
import jax.numpy as jnp

from jax import config
config.update("jax_enable_x64", True)


# Optimization of 1D SIGMOID shape
tol = None#1e-12 # ignored by jax.scipy.optimize.minimize

def sigmoid(x, parameters, erf=scipy_erf):
    return 0.5 * (erf(x / parameters[0]) + 1) * (parameters[1] - parameters[2]) + parameters[2]

x = jnp.arange(-2, 2, 0.001)
parameters_ref = np.array((2, # blur - sigma / 2**0.5
                           0.1, # f(x) for x > 0
                           0.9  # f(x) for x < 0
                           ))
y_gt = sigmoid(x, parameters_ref)

parameters_initial = np.array((0.25, 0.7, 0.2))
y_init = sigmoid(x, parameters_initial)

results = []
for mode in ("NUMPY", "JAX", "MIX"):
    def cost(params):
        if mode == "NUMPY":
            delta = sigmoid(x, params, scipy_erf) - sigmoid(x, parameters_ref, scipy_erf)
            return np.average(delta * delta)
        elif mode in ("JAX", "MIX"):
            delta = sigmoid(x, params, jax_erf) - sigmoid(x, parameters_ref, jax_erf)
            return jnp.average(delta * delta)
        else:
            raise ValueError("UPS")
    cost_jit = jax.jit(cost)
    cost_with_grad_jit = jax.jit(jax.value_and_grad(cost))
    
    @jax.jit
    def jax_minimize_this_cost_jit(x0):
        return jax_minimize(cost_jit, x0, method="BFGS", tol=tol)

    def opt_calc(x0):
        if mode == "NUMPY":
            res = scipy_minimize(cost, x0, method="BFGS", tol=tol)
        elif mode == "JAX":
            res = jax_minimize_this_cost_jit(x0)
        elif mode == "MIX":
            res = scipy_minimize(cost_with_grad_jit, x0, jac=True, method="BFGS", tol=tol)
        else:
            raise ValueError("UPS")
        return res
    
    print(f"{mode}:")
    %timeit opt_calc(parameters_initial)
    opt_res = opt_calc(parameters_initial)
    if opt_res.status != 0:
        raise ValueError("Ups, no convergence!")

    y_est = sigmoid(x, opt_res.x)
    solution_tolerance = np.abs(opt_res.x - parameters_ref)
    solution_tolerance /= parameters_ref + 1e-6
    desc = f"{mode} status: {opt_res.status} in {opt_res.nit}\nParams estimated: {opt_res.x}\nTolerance: {solution_tolerance}"

    results.append((y_est, desc))
    #print(f"{desc}\nHessian:\n{res.hess_inv}")

fig, ax = plt.subplots(layout="constrained")
fig.set_size_inches(18,8)
ax.plot(x[::100], y_gt[::100], "*", label="Ground Truth")
ax.plot(x, y_init, "--", label="Initial")
for y_est, desc in results:
    ax.plot(x, y_est, "--", label=desc)
ax.legend()
_ = ax.grid(True)

```
